File: orchestrator-api/src/main/java/com/sequenceiq/cloudbreak/orchestrator/GatewayConfig.java
package com.sequenceiq.cloudbreak.orchestrator;

public class GatewayConfig {

    private String publicAddress;
    private String privateAddress;
    private String certificateDir;

    public GatewayConfig(String publicAddress, String privateAddress, String certificateDir) {
        this.privateAddress = privateAddress;
        this.publicAddress = publicAddress;
        this.certificateDir = certificateDir;
    }

    public String getPublicAddress() {
        return publicAddress;
    }

    public String getPrivateAddress() {
        return privateAddress;
    }

    public String getCertificateDir() {
        return certificateDir;
    }
}

File: orchestrator-api/src/main/java/com/sequenceiq/cloudbreak/orchestrator/model/GatewayConfig.kt
package com.sequenceiq.cloudbreak.orchestrator.model

class GatewayConfig(val publicAddress: String, val privateAddress: String, val hostname: String?,
                    val gatewayPort: Int?, val certificateDir: String, val serverCert: String?, val clientCert: String?, val clientKey: String?) {

    constructor(publicAddress: String, privateAddress: String, gatewayPort: Int?, certificateDir: String) : this(publicAddress, privateAddress, null, gatewayPort, certificateDir, null, null, null) {
    }

    val gatewayUrl: String
        get() = String.format("https://%s:%d", publicAddress, gatewayPort)

    override fun toString(): String {
        return "GatewayConfig{"
        +"publicAddress='" + publicAddress + '\''
        +", privateAddress='" + privateAddress + '\''
        +", certificateDir='" + certificateDir + '\''
        +", serverCert='" + serverCert + '\''
        +", clientCert='" + clientCert + '\''
        +", clientKey='" + clientKey + '\''
        +'}'
    }
}
